test/unit/lineage.js

```diff
@@ -234,3 +320,137 @@ exports['minify minifies the contact and lineage'] = test => {
   test.deepEqual(actual, expected);
   test.done();
 };
+
+exports['hydrate+minify is noop on a report'] = test => {
+  const docId = 'abc';
+
+  const parentContact = {
+    _id: 'contact-def',
+    phone: '+123'
+  };
+  const grandparentContact = {
+    _id: 'contact-ghi',
+    phone: '+456'
+  };
+  const parent = {
+    _id: 'def',
+    name: 'jack',
+    contact: { _id: parentContact._id },
+    parent: {
+      _id: 'ghi'
+    }
+  };
+  const grandparent = {
+    _id: 'ghi',
+    contact: { _id: grandparentContact._id },
+    name: 'jim'
+  };
+  const givenContact = {
+    _id: 'contact',
+    name: 'apex',
+    parent: {
+      _id: parent._id,
+      parent: {
+        _id: grandparent._id
+      }
+    }
+  };
+  const given = {
+    _id: docId,
+    type: 'data_record',
+    form: 'A',
+    contact: {
+      _id: givenContact._id,
+      parent: {
+        _id: parent._id,
+        parent: {
+          _id: grandparent._id
+        }
+      }
+    }
+  };
+  sinon.stub(db.medic, 'view').callsArgWith(3, null, { rows: [
+    { doc: given },
+    { doc: givenContact },
+    { doc: parent },
+    { doc: grandparent }
+  ] });
+
+  const fetch = sinon.stub(db.medic, 'fetch');
+  fetch.callsFake((options, callback) => {
+    const contactDocs = options.keys.map(id => {
+      switch (id) {
+        case givenContact._id:
+          return givenContact;
+        case parentContact._id:
+          return parentContact;
+        case grandparentContact._id:
+          return grandparentContact;
+      }
+    });
+    callback(null, { rows: contactDocs.map(doc => { return {doc: doc}; }) });
+  });
+
+  lineage.fetchHydratedDoc(docId, (err, actual) => {
+    test.equals(err, null);
+    lineage.minify(actual);
+    test.deepEqual(given, actual);
```

Comment:
I'm not convinced this is testing correctly because minify modifies the given object. I _think_ it's ok because `fetchHydratedDoc` is returning a new object. It's safer however to clone `given` (eg: `JSON.parse(JSON.stringify(given))`) before using it to prove that it's actually working and not just referring to the same memory location and therefore will always be deepEqual...